### ISSUE TITLE
[BUG] Node names with `-` when using Vagrant fails

### ISSUE THREAD

**Issue by SudarshanVK** [ipspace/netlab#1499]
### To Reproduce

The Bug can be reproduced by trying to spin up the following topology

---
name: testing
provider: libvirt
defaults.device: cumulus_nvue

nodes:
  router_1:
    image: CumulusCommunity/cumulus-vx:5.10.0
  router_2:
    image: CumulusCommunity/cumulus-vx:5.10.0

links:
  - router-1:
      ifindex: 1
    pool: l2only
    name: Unused link
  - router-2:
      ifindex: 1
    pool: l2only
    name: Unused link
  - router-1:
      ifindex: 2
    router-2:
      ifindex: 2
    name: Link between router1-router2 

### Expected behavior

The '-' should be replaced when template out the vagrant file.

### Lab topology

```
name: testing
provider: libvirt
defaults.device: cumulus_nvue

nodes:
  router_1:
    image: CumulusCommunity/cumulus-vx:5.10.0
  router_2:
    image: CumulusCommunity/cumulus-vx:5.10.0

links:
  - router-1:
      ifindex: 1
    pool: l2only
    name: Unused link
  - router-2:
      ifindex: 1
    pool: l2only
    name: Unused link
  - router-1:
      ifindex: 2
    router-2:
      ifindex: 2
    name: Link between router1-router2 
```

### Output

`netlab create` does not generate any errors. However there is an error when spinning up the lab

```
Vagrant failed to initialize at a very early stage:

There is a syntax error in the following Vagrantfile. The syntax error
message is reproduced below for convenience:

/home/svk/all_labs/networklab/Vagrantfile:8: syntax error, unexpected '-', expecting '|'
...m.define "router-1" do |router-1|
...                              ^
/home/svk/all_labs/networklab/Vagrantfile:9: syntax error, unexpected symbol literal, expecting `end'
    router-1.vm.provider :libvirt do |domain|
                         ^
Error executing vagrant destroy -f:
  Command '['vagrant', 'destroy', '-f']' returned non-zero exit status 1.
[FATAL]   netlab down: vagrant destroy -f failed, aborting...
```

### Version

* netlab version 1.9.1

**Comment by SudarshanVK** [comment]
Solved this issue by modifying the file `netsim/templates/provider/libvirt/define-domian.j2`

Modify `{% set name = name|replace('-','') %}` to `{% set name = name | replace('-','') %}`

This might be due to some changes in the jinja2 package. I am using `Jinja2==3.1.4`

**Comment by ipspace** [comment]
This should be fixed by https://github.com/ipspace/netlab/commit/c8b16029fa97a573fe28cee11d3bf436456499f9 in release 1.9.2. Could you please upgrade and check?

Thank you!

**Comment by SudarshanVK** [comment]
Verified that this is indeed fixed in version 1.9.2.
Many thanks for the response. ❤️

**Comment by SudarshanVK** [comment]
Issue has been fixed in version 1.9.2
